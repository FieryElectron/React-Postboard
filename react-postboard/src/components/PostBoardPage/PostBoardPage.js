import Header from '../Header/Header';
import CreatePostField from '../CreatePostField/CreatePostField';
import PageSelector from '../PageSelector/PageSelector';
import PostList from '../PostList/PostList';
import TimeSelector from '../TimeSelector/TimeSelector';

import { useNavigate } from 'react-router-dom';
import Button from '../Button/Button';

const PostBoardPage = () => {
    const navigate = useNavigate();

    return (
        <div>
            <Header title='PostBoard' />
            <CreatePostField />
            <TimeSelector/>
            <PageSelector />
            <PostList />
            <Button text='Back' onClick={()=>{navigate("/")}}/>
        </div>
    )
}


export default PostBoardPage;