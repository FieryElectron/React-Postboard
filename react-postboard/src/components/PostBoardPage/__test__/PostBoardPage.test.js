import ReactDOM from "react-dom/client";
import { render, fireEvent  } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import '@testing-library/jest-dom/extend-expect';

import PostBoardPage from '../PostBoardPage';
import { BrowserRouter as Router} from 'react-router-dom'

let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

it("renders postBoardPage without crashing", ()=>{
    act(() => {
        ReactDOM.createRoot(container).render(<Router><PostBoardPage/></Router>);
    });
});


it("matchs snapshot 1", ()=>{
  const tree = renderer.create(<Router><PostBoardPage/></Router>).toJSON();
  expect(tree).toMatchSnapshot();
});
