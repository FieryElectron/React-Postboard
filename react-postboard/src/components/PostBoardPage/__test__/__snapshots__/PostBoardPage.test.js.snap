// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matchs snapshot 1 1`] = `
<div>
  <div
    className="headerContainer"
    data-testid="headerContainer"
  >
    <h1
      className="header"
      data-testid="header"
    >
      PostBoard
    </h1>
  </div>
  <div
    className="createPostFieldContainer"
    data-testid="createPostFieldContainer"
  >
    <div
      className="inputContainer"
      data-testid="inputContainer"
    >
      <input
        className="input"
        data-testid="input"
        placeholder="Post Content"
        type="type"
      />
    </div>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#38DD38",
        }
      }
    >
      Create
    </button>
  </div>
  <div
    className="timeSelectorContainer"
    data-testid="timeSelectorContainer"
  >
    <h3
      className="timeSelectorTitle"
      data-testid="timeSelectorTitle1"
    >
      From
    </h3>
    <input
      className="timeSelectorDate"
      data-testid="timeSelectorDate1"
      type="date"
    />
    <h3
      className="timeSelectorTitle"
      data-testid="timeSelectorTitle2"
    >
      To
    </h3>
    <input
      className="timeSelectorDate"
      data-testid="timeSelectorDate2"
      type="date"
    />
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#0071FE",
        }
      }
    >
      Refresh Posts
    </button>
  </div>
  <div
    className="pageSelectorContainer"
    data-testid="pageSelectorContainer"
  >
    <span
      className="pageSelectorTitle"
      data-testid="pageSelectorTitle"
    >
      Page
    </span>
    <select
      className="pageSelectorSelect"
      data-testid="pageSelectorSelect"
    >
      <option
        value="0"
      >
        0
      </option>
      <option
        value="1"
      >
        1
      </option>
      <option
        value="2"
      >
        2
      </option>
      <option
        value="3"
      >
        3
      </option>
      <option
        value="4"
      >
        4
      </option>
      <option
        value="5"
      >
        5
      </option>
    </select>
  </div>
  <div>
    <div
      className="singlePostContainer"
      data-testid="singlePostContainer"
    >
      <div
        className="ownerName"
        data-testid="ownerName"
      >
        ownerName
      </div>
      <div
        className="postContent"
        data-testid="postContent"
      >
        Content
      </div>
      <hr />
      <div
        className="timeStampContainer"
        data-testid="timeStampContainer"
      >
        <span
          className="timeStamp"
          data-testid="timeStamp"
        >
          timestamp
        </span>
        <button
          className="button"
          data-testid="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#E62E2E",
            }
          }
        >
          Delete
        </button>
      </div>
    </div>
    <div
      className="singlePostContainer"
      data-testid="singlePostContainer"
    >
      <div
        className="ownerName"
        data-testid="ownerName"
      >
        ownerName
      </div>
      <div
        className="postContent"
        data-testid="postContent"
      >
        Content
      </div>
      <hr />
      <div
        className="timeStampContainer"
        data-testid="timeStampContainer"
      >
        <span
          className="timeStamp"
          data-testid="timeStamp"
        >
          timestamp
        </span>
        <button
          className="button"
          data-testid="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#E62E2E",
            }
          }
        >
          Delete
        </button>
      </div>
    </div>
    <div
      className="singlePostContainer"
      data-testid="singlePostContainer"
    >
      <div
        className="ownerName"
        data-testid="ownerName"
      >
        ownerName
      </div>
      <div
        className="postContent"
        data-testid="postContent"
      >
        Content
      </div>
      <hr />
      <div
        className="timeStampContainer"
        data-testid="timeStampContainer"
      >
        <span
          className="timeStamp"
          data-testid="timeStamp"
        >
          timestamp
        </span>
        <button
          className="button"
          data-testid="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#E62E2E",
            }
          }
        >
          Delete
        </button>
      </div>
    </div>
  </div>
  <button
    className="button"
    data-testid="button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#0071FE",
      }
    }
  >
    Back
  </button>
</div>
`;
